source/models/table.py

```diff
@@ -47,22 +53,65 @@ def set(self, row):
             old_row.delete()
         row_obj.serialize()
 
-    
-    def get(self):
-        pass
-    
+
+    def get_similar_primarykeys(self, query):
+        primarykeys = []
+        for index_name in query:
+            if index_name in self.table_metadata.index_keys:
+                primarykeys = Index(self, index_name = index_name, index_value = query[index_name]).get_primary_key(index_value = query[index_name])
+                break
+        return primarykeys
+ 
+
+    def get(self, query):
+        self.__colomns_name_validate(query)
+        primarykeys = []
+        if Keys.PRIMARY_KEY in query : # if primarykey in the query
+            primarykeys.append(query[Keys.PRIMARY_KEY])     
+        else:
+            primarykeys = self.get_similar_primarykeys(query = query)
+            if not len(primarykeys):# compare all data in the table with the query
+                primarykeys = self.get_all_primary_keys()
+        row_objects = self.get_rows(primarykeys)
+        return self.filter_by_query(row_objects = row_objects, query = query)
+
+
+    def get_rows(self, primarykeys):
+            row_objects = []
+            for primarykey in primarykeys:
+               row_objects.append(Row(self).load_by_primary_key(self, primarykey))
+            return row_objects
+
+
+    def filter_by_query(self, row_objects, query):
+        filtered_objects = []
+        for row_object in row_objects:
+            if Table.compare(query = query, row_object = row_object):
```

Comment:
I would suggest having a method inside the row class called `self.has_attributes(query)` instead of Table.compare

Reply:
i could made it by why ? what is the problem with table.compare function?

Reply:
ok I put it in raw class and made the value private, and also made a getter function to take the value